bigchaindb/core.py

```diff
@@ -174,36 +174,38 @@ def get_tx_by_payload_hash(self, payload_hash):
             returns `None`
         """
 
-        cursor = r.table('bigchain')\
-            .get_all(payload_hash, index='payload_hash')\
+        cursor = r.table('bigchain') \
+            .get_all(payload_hash, index='payload_hash') \
             .run(self.conn)
 
         transactions = list(cursor)
         return transactions
 
-    def get_spent(self, txid):
+    def get_spent(self, tx_input):
         """Check if a `txid` was already used as an input.
 
         A transaction can be used as an input for another transaction. Bigchain needs to make sure that a
         given `txid` is only used once.
 
         Args:
-            txid (str): transaction id.
+            tx_input (dict): Input of a transaction in the form `{'txid': 'transaction id', 'cid': 'condition id'}`
 
         Returns:
             The transaction that used the `txid` as an input if it exists else it returns `None`
         """
         # checks if an input was already spent
-        # checks if the bigchain has any transaction with input `transaction_id`
+        # checks if the bigchain has any transaction with input {'txid': ..., 'cid': ...}
         response = r.table('bigchain').concat_map(lambda doc: doc['block']['transactions'])\
```

Comment:
what's the need of `lamba` here?
`r.table('bigchain').concat_map(r.row['block']['transactions']) ...` is much simpler and should be equivalent.